app/components/answer-step.jsx

```diff
@@ -0,0 +1,152 @@
+// https://github.com/EnCiv/civil-pursuit/issues/102
+
+// most of this code was taken from why-step.jsx
+'use strict'
+import React, { forwardRef, useState, useEffect } from 'react'
+import StepIntro from './step-intro'
+import WhyInput from './why-input'
+import cx from 'classnames'
+import { createUseStyles } from 'react-jss'
+import _ from 'lodash'
+
+const AnswerStep = forwardRef((props, ref) => {
+  const {
+    className = '',
+    intro = '',
+    question = {},
+    whyQuestion = '',
+    shared = {
+      startingPoint: {},
+      whyMosts: [],
+    },
+    onDone = () => {},
+    ...otherProps
+  } = props
+  const classes = useStylesFromThemeFunction()
+
+  // user input to answer important issue
+  const [startingPoint, setStartingPoint] = useState(shared.startingPoint)
+
+  // user input to explain why issue is important
+  const [whyMosts, setWhyMosts] = useState(shared.whyMosts)
+
+  useEffect(() => {
+    if (areStartingComplete() && areWhyAnswersComplete()) {
+      onDone({ valid: true, value: { startingPoint: startingPoint, whyMosts: whyMosts } })
+    } else {
+      onDone({ valid: false, value: { startingPoint: startingPoint, whyMosts: whyMosts } })
+    }
+  }, [whyMosts])
+
+  const updateQuestionResponse = ({ valid, value }) => {
+    const updatedAnswers = {
```

Comment:
> But based on the learnings of whyStep - we don't want to call setStartingPoint if the new value for starting point is coming up from the pointInput.

How exactly can I check this? Before I just set the value to whatever the user inputs into either startingPoint or whyMost, and it seems to work fine with tests (no infinite loop), so where should I put the code you provided in my code?

Reply:
The loop to worry about happens when new data comes down from the database after the component is first rendered.  The code should be in the onDone handler from the child.

Reply:
Do i have to use pointByPart when useEffect() is called then? Everything works fine as is but when I try extracting the answer and why fields into the onDone in the useEffect method nothing updates anymore. 

And I just wanted to clarify, is the only reason pointByPart is needed is to prevent infinite loops when the page first loads in? And then startingPoint and whyMosts takes care of user inputs afterwards?

Reply:
after you call onDone from this component, it should display the results at the bottom of the component in storybook.  You may need to add the decorator for that, see the story for whyinput componet. 

onDone should be called from within use effect when the component first renders, but other wise it should be called from with the handlers for onDone to the whyInput component.

Reply:
Gotcha. So do I still use pointByPart to update the state with onDone or do I leave it as is and use startingPoint and whyMost?

Reply:
https://github.com/user-attachments/assets/a6705e4c-c2d7-4751-b653-88dc8c5d8070

There is a super weird bug I am getting when extracting info with pointByPart. Video attached shows inconsistencies when updating data. I think we should just stick with updating onDone with startingPoint and whyMost, but let me know what you think and if you can help with the bug.

Also another thing in the video was that the original bug (answer not updating immeadiately until after why is updated) is still there.